starky/src/f5g.rs

```diff
@@ -150,35 +182,107 @@ impl F5G {
             return Self::ONE;
         }
 
-        let mut res = self;
+        let mut res = *self;
         for i in (0..bits.len() - 1).rev() {
             res.square();
             if bits[i] == 1 {
-                res = res.mul(self);
+                res = res.mul(*self);
             }
         }
         res
     }
 
+    // Invert this element. If this value is zero, then zero is returned.
+    // Inv() function refers to the implementation of ecgfp5: https://github.com/pornin/ecgfp5/blob/ce059c6d1e1662db437aecbf3db6bb67fe63c716/python/ecGFp5.py#L751
     #[inline]
-    pub fn batch_inverse(elems: &[Self]) -> Vec<Self> {
-        if elems.len() == 0 {
-            return vec![];
-        }
+    fn inv(&self) -> Self {
+        match self.dim {
+            5 => {
```

Comment:
I think you can keep the old reverse function, but mark it with modifier #[inline(always)], such that the code would be more friendly to the reader.

Reply:
I understand what you mean by keeping the old code like the one below
```
fn inv(&self) -> Self {
    self._inv()
}
```

Reply:
yes, exactly